File: frontend/quiz/src/seleniumTests/Pages/ChooseNumberPage.js
import Page from "./Page";
import { getElementById } from "./Utility";
class ChooseNumberPage extends Page {
  get Header() {
    return getElementById(this.driver, "header");
  }
  get startTheQuizBtn() {
    return getElementById(this.driver, "startButton");
  }
  get numberOfQuestionsInput() {
    return getElementById(this.driver, "numberOfQuestionsInput");
  }
  async open(path) {
    await super.open(path);
  }
}

export default ChooseNumberPage;

File: backend/selenium/ChooseNumberPage.py
from Utility import getElementById
from Page import Page
headerId = "header"
startTheQuizBtnId = "startButton"
numberOfQuestionsInputID = "numberOfQuestionsInput"


class ChooseNumberPage(Page):
    def Header(self):
        return getElementById(self.driver, headerId)

    def startTheQuizBtn(self):
        return getElementById(self.driver, startTheQuizBtnId)

    def numberOfQuestionsInput(self):
        return getElementById(self.driver, numberOfQuestionsInputID)

    def open(self):
        super.open("choosesNumber")
